### PR TITLE
Mob coding session 2022-09-20

### PR DESCRIPTION
**Pull request by aaronsteers**
Note:
- 'TODO:' marks areas that cannot be merged as is

Co-authored-by: Will Da Silva <will@willdasilva.xyz>
Co-authored-by: Edgar R. M. <edgarrmondragon@users.noreply.github.com>
Co-authored-by: Cody J. Hanson <cody@codyjhanson.com>
Co-authored-by: Reina Feather <feather.rw@gmail.com>

**[Deploy Preview](https://deploy-preview-813--meltano-hub-gridsome.netlify.app/)**

In Scope:

- [ ] #784 - @WillDaSilva 
- [ ] #792 - @aaronsteers 
	- may be a partial solve
- [ ] #779 - @edgarrmondragon 
- [ ] #773 - @rwfeather 
- [ ] #778 - @cjohnhanson

### CODE REVIEW

**Comment by edgarrmondragon** [comment]
Build error:

```
TypeError: Cannot read properties of undefined (reading 'keywords')
```

that's on my changes most likely. Looking...

File: src/components/PluginSidebar.vue [hunk 1]
```diff
@@ -94,7 +98,19 @@
       </ul>
     </div>
 
-    <div><p class="text-lg">Maintainer</p></div>
+    <div>
+      <p class="text-lg">Maintainer</p>
+      <ul v-if="variant.node.keywords.includes('meltano_sdk')" class="list-disc list-inside shields">
```
Comment (edgarrmondragon) [inline src/components/PluginSidebar.vue:103]:
```suggestion
      <ul v-if="(keywords ?? []).includes('meltano_sdk')" class="list-disc list-inside shields">
```

File: src/components/PluginSidebar.vue [hunk 2]
```diff
@@ -94,7 +98,19 @@
...
-    <div><p class="text-lg">Maintainer</p></div>
+    <div>
+      <p class="text-lg">Maintainer</p>
+      <ul v-if="(keywords ?? []).includes('meltano_sdk')" class="list-disc list-inside shields">
+        <li>
+          <a href="https://sdk.meltano.com/en/latest/">
+            <img
+              alt="Built with the Meltano SDK"
+              src="https://img.shields.io/badge/-Built%20with%20the%20Meltano%20SDK-blueviolet"
+            >
```
Comment (edgarrmondragon) [inline src/components/PluginSidebar.vue:109]:
```suggestion
            />
```

**Comment by cjohnhanson** [comment]
@rwfeather @aaronsteers it looks like since preflight makes headings and lists unstyled by default, our rended markdown for things like `next_steps` doesn't get styled and looks a little rough. Any ideas on a quick-ish fix for this? I have two not-so-great ideas for high-effort fixes that we could do, but was hoping there's an easier way. My bad ideas:
- Rewrite all of our markdown in plugin definitions as html with proper styling. *This will also make contributing new plugins to the hub more difficult*
- Try to standardize things like `next_steps` in plugin definitions so that we can add the styles inline ourselves. e.g. we might have `steps:` as a child key of `next_steps` and then we'd `v-for` through each step and add a styled list element for it.

File: src/pages/singer/Docs.vue [hunk 3]
```diff
@@ -1,358 +1,371 @@
...
-    "subcol_1": 1,
-    "subcol_2": 2
+    <div class="grid w-full">
+      <div class="h-7 max-h-7"></div>
+      <div class=" bg-white border text-black w-full prose lg:prose-xl">
+        ## Test test
+
+        This is markdown, `kinda`.
+
+        TODO: See if we can render from raw markdown here, or from a file. (Doesn't render as of now.)
```
Comment (aaronsteers) [inline src/pages/singer/Docs.vue:10]: - [ ] TODO: @aaronsteers

File: src/pages/singer/Spec.vue [hunk 4]
```diff
@@ -1,238 +1,29 @@
...
-    messages from stdin, but typically they do not do anything with the state messages beyond
-    sending them to stdout. This is done once all data that appeared in the stream before the state
-    message has been processed by the Target. Schema Files # Targets do not take a schema file.
-    However, they are expected to read the schema messages from stdin and perform validation of the
-    incoming record using the provided schema.
+    <div class="grid w-full">
+      <div class="h-7 max-h-7"></div>
+      <div class=" bg-white border text-black w-full">
+        <p class="text-3xl py4">Singer Spec</p>
+        TODO: Add rendered content here from <code>/src/pages/singer/spec_content.md</code>
```
Comment (aaronsteers) [inline src/pages/singer/Spec.vue:7]: - [ ] TODO: @aaronsteers

**Review by aaronsteers — COMMENTED** [review]
Just two action items on my side and then I think this is ready to go.

- @cjohnhanson and @rwfeather - if you have time today can you add comments and approve by end of today if possible? Conditional approval is also okay if you want to call out what is missing.

**Review by cjohnhanson — APPROVED** [review]
@aaronsteers I think besides those pending action items, this should be good.

File: src/pages/singer/Docs.vue (hunk 3 above)
Comment (aaronsteers) [reply src/pages/singer/Docs.vue]: Resolved in 3012d832dbbce3d33564479baf9b0e0909fdc3c8 and aee1f4cead0db1de8d7d8d26d01337642889592f.

File: src/pages/singer/Spec.vue (hunk 4 above)
Comment (aaronsteers) [reply src/pages/singer/Spec.vue]: Ditto. Resolved in 3012d832dbbce3d33564479baf9b0e0909fdc3c8 and aee1f4cead0db1de8d7d8d26d01337642889592f.

**Comment by aaronsteers** [comment]
Thanks, @cjohnhanson. Moving forward with merge.

@cjohnhanson, @rwfeather, @edgarrmondragon, @tayloramurphy, @WillDaSilva:

Markdown pages will now be automatically rendered to html if in paths like `/static/markdown/**/*.md`. The path will match the relative path from `/static/markdown/`. Markdown rendering is performed by the `MarkdownDocs` template. 

More info and developer instructions for markdown rendering have been added to `CONTRIBUTING.md`.

**Comment by aaronsteers** [comment]
Known issues as of now with markdown page rendering:

- [ ] Syntax highlighting for code blocks does not appear to be working as of yet.
- [ ] The content `div` is left-aligned, wheras it should instead be either full-width or centered with a border on either side.
- [ ] We should add back borders as/when we're able.
- [ ] We should add back the right-hand TOC nav as/when we're able.

**Pull request merged**